src/NuGet.Clients/NuGet.CommandLine/Commands/InstallCommand.cs

```diff
@@ -352,9 +353,13 @@ private DependencyBehavior GetDependencyBehavior()
             }
             else
             {
+                var signedPackageVerifier = new SignedPackageVerifier(
+                            SignatureVerificationProviderFactory.GetSignatureVerificationProviders(),
+                            SignedPackageVerifierSettings.Default);
+
                 var projectContext = new ConsoleProjectContext(Console)
                 {
-                    PackageExtractionContext = new PackageExtractionContext(Console)
+                    PackageExtractionContext = new PackageExtractionV2Context(Console, signedPackageVerifier)
```

Comment:
Why add a new class if you're removing the PackageExtractionContext regardless?
Why not just rename it?

Reply:
Good point, I should change the class directly

Reply:
nvm...I just noticed that there's a PackageExtractionV3Context as well. 

I am not fan of this type of naming as it does not provide any additional meaning. 

Does V2/V3 have anything to do with the protocols themselves? 

It's just very confusing imo.